[gnoossk] iP

gnoossk opened the pull request:
# Hue 
> Believe you can and you're halfway there. - Theodore Roosevelt

A __cool__ new bot to keep track of things to do! It 
* Keeps track of your tasks, deadlines and events coming up
* lists them in a comprehensive manner
* has its own flair 

Here is the step-by-step guide to getting the app:

1. [Go to the source code](https://github.com/thony-ui/ip).
2. Click on Releases.
3. Download the JAR file.
4. Open your favorite IDE (we recommend IntelliJ).
5. Open the JAR file and run the code 🎉.
6. And the best part? It’s FREE! 😊

Features 
- [x] Managing Tasks
- [x] Managing Deadlines
- [x] Managing Events
- [ ] Comprehensive UI (Coming Soon)


Here is a snippet of the entry program in `Main`

```java
public class Main {
    public static void main(String[] args) {
        new Hue("./data/hue.txt").run();
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

gnoossk committed:
Finish level 0

gnoossk committed:
Finish Level1-1

gnoossk committed:
Finish Level 1-2

gnoossk committed:
Add Task.java

gnoossk committed:
Finish Level1-3

gnoossk committed:
Add Todo, Event and Deadline Classes

gnoossk committed:
Finish Level 1-4

gnoossk committed:
Reformat Hue.java and handle known exceptions

gnoossk committed:
Finish Level1-4

gnoossk committed:
Revert "Finish Level1-4"

This reverts commit f54993478c6347b21242eebffdbb7a318f91a381.

gnoossk committed:
Finish Level1-5

gnoossk committed:
Add delete function and change input.txt

gnoossk committed:
Finish Level1-6

gnoossk committed:
Create Storage.java

gnoossk committed:
Add toFileFormat() methods to Tasks

gnoossk committed:
Add LoadTasks() and CreateTaskFromData() to Storage.java

gnoossk committed:
Edit Hue.java and Storage.java for File Read and Write

gnoossk committed:
Merge branch 'branch-level-7'

gnoossk committed:
Remove Hue.txt

gnoossk committed:
Edit toFileFormat methods in Deadline and Event

gnoossk committed:
Merge branch 'branch-level-7'

gnoossk committed:
Add Date and Time formatting to Event and Deadline

gnoossk committed:
Change variable names for Date/Time formatting

gnoossk committed:
Merge branch 'branch-level-8'

gnoossk committed:
Remodeled chatBot incorporating OOP elements

gnoossk committed:
Merge branch 'branch-A-More-OOP'

gnoossk committed:
Add JavaDocs and Follow Coding Standard

gnoossk committed:
Merge branch 'branch-A-Packages'

gnoossk committed:
Merge branch 'add-gradle-support'

gnoossk committed:
Add JUnit Tests

gnoossk committed:
change build.gradle for Hue.jar

gnoossk committed:
Create FindCommand.java

gnoossk committed:
Add properly working Find Command

gnoossk committed:
Merge branch 'branch-level-9'

gnoossk committed:
Revert "Add properly working Find Command"

This reverts commit eac358575d874faf7a9178f33afed3891d34c867.

gnoossk committed:
Merge branch 'branch-level-9'

gnoossk committed:
Re-implement Find

gnoossk committed:
Revert "Re-implement Find"

This reverts commit e611c472c86004070947a501dc69bac025c63eff.

gnoossk committed:
edit Task.java

gnoossk committed:
Implement working Find

gnoossk committed:
Merge branch 'branch-level-9'

gnoossk committed:
De-conflict master and branch-level-9

gnoossk committed:
Merge branch 'branch-level-9'

gnoossk committed:
Fix Bug with Event.java

gnoossk committed:
Merge branch 'branch-level-9'

gnoossk committed:
Edit Hue.txt

gnoossk committed:
Merge branch 'master' into branch-A-JavaDoc

gnoossk committed:
edit hue.txt

gnoossk committed:
Merge branch 'master' into branch-A-CodingStandard

gnoossk committed:
edit Hue.txt

gnoossk committed:
Merge branch 'branch-A-JavaDoc'

gnoossk committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	data/Hue.txt

gnoossk committed:
Add JavaFX

gnoossk committed:
Reformat Hue to work with the new javaFX GUI

gnoossk committed:
Improve UI of Hue ChatBot and Check Style of Code

gnoossk committed:
Merge branch 'branch-level-10'

gnoossk committed:
Attempt varags

gnoossk committed:
Merge branch 'branch-A-Varags'

gnoossk committed:
Checkstyle of all scripts

gnoossk committed:
Merge branch 'branch-A-Checkstyle'

Bonzzz3 commented on src/main/java/hue/UI/UI.java:
```diff
@@ -0,0 +1,76 @@
+package hue.UI;
```
Names representing packages should be in all lower case according to Java coding standard.

Bonzzz3 commented on src/main/java/hue/UI/UI.java:
```diff
@@ -0,0 +1,76 @@
+/**
+ * Handles interactions with the user through the command line interface.
+ */
+public class UI {
```
I think the class name should be Ui or something like UserInterface instead since UI is not really in PascalCase.

Bonzzz3 commented on src/main/java/hue/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,51 @@
+package hue.command;
```
commands could be a better package name.

Bonzzz3 commented on src/main/java/hue/task/Todo.java:
```diff
@@ -0,0 +1,29 @@
+/**
+ * Represents a todo task.
+ */
+public class Todo extends Task {
```
Hmm I'm thinking ToDo could be a better class name.

Bonzzz3 commented on src/main/java/hue/command/AddTodoCommand.java:
```diff
@@ -0,0 +1,44 @@
+/**
+ * Represents a command to add a todo task.
+ */
+public class AddTodoCommand extends Command {
```
Maybe AddToDoCommand could be a better class name

Bonzzz3 commented on src/main/resources/images/DaUser.png:
Nice png hahaha

Bonzzz3 commented on src/main/java/hue/Hue.java:
```diff
@@ -0,0 +1,77 @@
+package hue;
+
+import hue.command.*;
```
Imported classes should always be listed explicitly. Sadly our IDE automatically include all files (*) :(

Bonzzz3 commented on src/main/java/hue/UI/UI.java:
```diff
@@ -0,0 +1,76 @@
+
+import java.util.ArrayList;
+import java.util.Scanner;
+import hue.task.*;
```
Refer to previous comment. (No *)

Bonzzz3 commented on src/main/java/hue/command/Command.java:
```diff
@@ -0,0 +1,39 @@
+
+import java.io.IOException;
+import hue.UI.UI;
+import hue.task.*;
```
Refer to previous comment. (No *)

Bonzzz3 commented on src/main/java/hue/parser/Parser.java:
```diff
@@ -0,0 +1,37 @@
+package hue.parser;
+
+import hue.command.*;
```
Refer to previous comment. (No *)

Bonzzz3 commented on src/main/java/hue/storage/Storage.java:
```diff
@@ -0,0 +1,114 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.File;
+import hue.task.*;
```
Refer to previous comment. (No *)

Bonzzz3 commented on src/test/java/hue/DateUtilsTest.java:
```diff
@@ -0,0 +1,41 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Refer to previous comment. (No *)

Bonzzz3 reviewed (commented):
I can't find many coding standard violation. Layout is nicely done. Great job! 👍

gnoossk committed:
Add Assertions to Relevant Files

gnoossk committed:
Fix Code Quality of Parser.java

gnoossk committed:
Merge branch 'branch-A-Assertions'

gnoossk committed:
Merge branch 'master' into branch-A-CodeQuality

gnoossk committed:
Merge branch 'branch-A-CodeQuality'

gnoossk committed:
Refactor naming of the ui package

There are several classes that use the ui package. Importing "hue.UI"
did not follow the coding standards as all caps breaks the convention
that only constants should use this naming style

As such, IntelliJ's Refactoring was used to follow conventional
guidelines

gnoossk committed:
Add gradle.yml

gnoossk committed:
Reformat gradle.yml

gnoossk committed:
Merge branch 'master' into branch-A-Personality

gnoossk committed:
Add personality flairs to the UI

gnoossk committed:
Merge pull request #1 from gnoossk/branch-A-Personality

Branch a personality

gnoossk committed:
Merge branch 'branch-A-Personality'

gnoossk committed:
Merge branch 'master' of https://github.com/gnoossk/ip

gnoossk committed:
Add Reschedule Command

gnoossk committed:
Add Reschedule Command to Hue

gnoossk committed:
Merge branch 'branch-B-snooze'

gnoossk committed:
Improve overall quality of code

Bug fixed where MMM-dd-yyyy format was not working when rescheduling
Events

Add better GUI, personality, more error handling and more testing

gnoossk committed:
Update README for User Guide

gnoossk committed:
Add proper file creation for isolated jar file and submit final version

gnoossk committed:
Edit User Guide

gnoossk committed:
Fix Bug with Inputs

gnoossk committed:
Merge branch 'fix-NPE'

gnoossk committed:
Fix DateParse Bug

gnoossk committed:
fix error message not appearing in UI

gnoossk committed:
Update User Guide

gnoossk committed:
fix UG and add Stage Header & Icon

gnoossk committed:
Fix mark command bug and update UG

gnoossk committed:
Merge branch 'fix-iP-bugs'

gnoossk committed:
Edit exception handling for new release

gnoossk committed:
Edit User Guide and Error Messages

gnoossk committed:
Reformat reschedule command and edit UG

gnoossk committed:
Merge branch 'reformat-reschedule'

gnoossk committed:
Update User Guide

gnoossk committed:
Update User Guide

gnoossk committed:
Update User Guide

gnoossk committed:
Fix Brace Style
